### PR TITLE
Feature/1408 add api tag

### PR DESCRIPTION
**Pull request by stereomon**
Add fixer for @api DocComment annotation

### CODE REVIEW

File: Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php [hunk 1]
```diff
@@ -0,0 +1,247 @@
+<?php
+
+namespace Spryker\Sniffs\Commenting;
+
+/**
+ * Checks if doc block contains @api annotation
```
Comment (dereuromark) [inline Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php]: which ones?. should be added that only the API relevant ones.

File: Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        return true;
+    }
+
+    /**
+     * @param \PHP_CodeSniffer_File $phpCsFile
+     * @param int $stackPointer
+     *
+     * @return string
+     */
+    protected function getNamespace(\PHP_CodeSniffer_File $phpCsFile, $stackPointer)
```
Comment (dereuromark) [inline Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php]: I am pretty sure this exists in the Abstract one already.

**Comment by dereuromark** [comment]
Cool, that we have tests again etc.
:+1: when all is cleared up

File: Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php (hunk 2 above)
Comment (stereomon) [inline Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php]: There is, but this one takes the filename and tries to get the class name and the namespace out of it. This is pretty old and must be checked if we can change it. This implementation is different, it searches for the namespace parts of the class. The current implementation also works with the current test files which would give a false namespace if i would use them.

Comment (dereuromark) [inline Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php]: ok

Comment (dereuromark) [inline Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php]: Using the namespace instead of file is always better, yes.

Comment (dereuromark) [inline Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php]: Actually, your implementation seems to problematic, as it does not include whitespaces and aliasing.
better to go all the way to the semicolon only.

You should be using getNamespace() of FullyQualifiedClassNameInDocBlockSniff (move it to abstract)
and use that one instead IMO.

Comment (stereomon) [inline Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php]: It is enough for what we need here, i only need the first two to three namespace elements

Comment (dereuromark) [inline Spryker/Sniffs/Commenting/DocBlockApiAnnotationSniff.php]: ah ok :+1:

**Pull request merged**